### DISCUSSION TITLE
Input size change after fed into HGT

### DISCUSSION THREAD

**Discussion by Gori-LV** [pyg-team/pytorch_geometric#4181] [Q&A]
Hi, I'm trying out HGT but having a problem of copied input size change. Below is the code I add to the end of [this](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hgt_dblp.py).

```python
import copy

x_dict, edge_index_dict = data.x_dict, data.edge_index_dict
copied_x_dict, copied_edge_index_dict = copy.deepcopy(data.x_dict), copy.deepcopy(data.edge_index_dict)

print(data.x_dict['author'].size())
print(x_dict['author'].size())
print(copied_x_dict['author'].size())

model.eval()
with torch.no_grad():
    _ = model(x_dict, edge_index_dict)
    _ = model(data.x_dict, data.edge_index_dict)
    _ = model(copied_x_dict, copied_edge_index_dict)

print(data.x_dict['author'].size())
print(x_dict['author'].size())
print(copied_x_dict['author'].size())

```
then got this output
```
torch.Size([4057, 451])
torch.Size([4057, 451])
torch.Size([4057, 451])
torch.Size([4057, 451])
torch.Size([4057, 64])
torch.Size([4057, 64])
```

Not sure where is the problem :( can someone help? Thx a lot!

**Comment by rusty1s** [comment]
I can't seem to reproduce this, see [this newly added test](https://github.com/pyg-team/pytorch_geometric/blob/master/test/nn/conv/test_hgt_conv.py#L126-L146). Which PyG version are you using?

**Reply by Gori-LV** [reply]
I tested the newly added test and got no error either, cuz `HGTConv` is not the problem. I figure that [this](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/hetero/hgt_dblp.py#L36) is the cause, when `x_dict` copied from `data.x_dict` is passed to the function, it is modified. But if `data.x_dict` is directly passed, it will fine (can you explain why?). 

I add `x_dict = copy.deepcopy(x_dict)` in the `forward()` function before `x_dict` is processed, the issue is solved.

**Reply by Gori-LV** [reply]
I suppose this may cause a waste of memory, is there a better way to do it? :(

**Reply by rusty1s** [reply]
Ah, I see. You are talking about `HGT` rather than `HGTConv`. This is easily fixable by replacing it with:
```python
for node_type, x in x_dict.items():
    x_dict[node_type] = self.lin_dict[node_type](x).relu_()
```
with
```python
x_dict = {key: self.lin_dict[key](x).relu_() for key, x in x_dict.items()}
```

**Reply by Gori-LV** [reply]
That's cool, thanks alot!
